src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodTranslator.cs

```diff
@@ -56,11 +56,8 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
                 return
                     _sqlExpressionFactory.GreaterThan(
                         _jsonTranslator.TranslateArrayLength(arrayOperand) ??
-                        _sqlExpressionFactory.Function(
-                            "array_length",
-                            new[] { arrayOperand, _sqlExpressionFactory.Constant(1) },
-                            typeof(int)),
-                        _sqlExpressionFactory.Constant(1));
+                        _sqlExpressionFactory.Function("cardinality", arguments, typeof(int?)),
```

Comment:
Change `int?` to `int` no? Cardinality never seems to return null...

Reply:
It returns `NULL` when the parameter is `NULL`.

Reply:
Ah of course.